dotCMS/src/main/java/com/dotmarketing/business/helper/PermissionHelper.java

```diff
@@ -0,0 +1,95 @@
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.PermissionAPI.PermissionableType;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotDataValidationException;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class for operations that involves the use of the {@link PermissionAPI}
+ */
+public class PermissionHelper {
+
+    private final PermissionAPI permissionAPI;
+
+    private PermissionHelper() {
+        this.permissionAPI = APILocator.getPermissionAPI();
+    }
+
+    private static class PermissionSingletonHelper {
+
+        private static final PermissionHelper INSTANCE = new PermissionHelper();
+    }
+
+    public static PermissionHelper getInstance() {
+        return PermissionHelper.PermissionSingletonHelper.INSTANCE;
+    }
+
+    /**
+     * Given a user and optional permissions and permissionableTypes, returns a map with the
+     * permissions allowed by permission type
+     *
+     * @param user User who requests the permissions info
+     * @param permissions Optional list with the permissions to be required (view {@link
+     * PermissionAPI#PERMISSION_READ} and {@link PermissionAPI#PERMISSION_WRITE}). If not set,
+     * {@link PermissionAPI#PERMISSION_READ} and {@link PermissionAPI#PERMISSION_WRITE} will be
+     * returned
+     * @param permissionableTypes Optional list of PermissionableType's names to query (view allowed
+     * values in {@link PermissionableType} enum). If not set, all values in the {@link
+     * PermissionableType} enum will be returned
+     * @return Map with the permissions allowed by permission type: Map<PermissionableType's name,
+     * Map<`canRead/canWrite`, Boolean>>
+     */
+    public Map<String, Map<String, Boolean>> getPermissionsByPermissionType(final User user,
+            final List<Integer> permissions, final List<String> permissionableTypes)
+            throws DotDataException {
+
+        if (null == user) {
+            throw new DotDataValidationException("User cannot be null");
+        }
+
+        final Map<String, Map<String, Boolean>> result = new HashMap<>();
+
+        final List<Integer> permissionsToReturn = UtilMethods.isSet(permissions) ? permissions
+                : CollectionsUtils
+                        .list(PermissionAPI.PERMISSION_READ, PermissionAPI.PERMISSION_WRITE);
+
+        final List<PermissionableType> permissionTypesToReturn =
+                UtilMethods.isSet(permissionableTypes) ? permissionableTypes.stream()
+                        .map(type -> PermissionableType.valueOf(type)).collect(
+                                Collectors.toList()) : Arrays.asList(PermissionableType.values());
```

Comment:
I think this Arrays.asList(PermissionableType.values())
Could be a constant

Reply:
Done